lib/pbench/agent/tool_meister_client.py

```diff
@@ -224,3 +228,76 @@ def terminate(self, group, interrupt=False):
                 )
                 ret_val = 1
         return ret_val
+
+
+def main(argv: list) -> int:
+    """Main program for the Tool Meister client CLI.  The command line
+    arguments are:
+
+      group - the tool group on which the actions will be taken
+
+      directory - the directory where data gathered from the actions will be
+                  stored
+
+      action - the particular action to take, can we one of "start", "stop",
+               or "send" (see `cli_tm_allowed_actions`).
+    """
+    PROG = os.path.basename(argv[0])
+    logger = logging.getLogger(PROG)
+
+    if os.environ.get("_PBENCH_TOOL_MEISTER_CLIENT_LOG_LEVEL") == "debug":
+        log_level = logging.DEBUG
+    else:
+        log_level = logging.INFO
+    logger.setLevel(log_level)
+    sh = logging.StreamHandler()
+    sh.setLevel(log_level)
+    shf = logging.Formatter(f"{PROG}: %(message)s")
+    sh.setFormatter(shf)
+    logger.addHandler(sh)
+
+    try:
+        group = argv[1]
+    except IndexError:
+        logger.error("Missing group argument")
+        return 1
+    try:
+        directory = argv[2]
+    except IndexError:
+        logger.error("Missing directory argument")
+        return 1
+    try:
+        action = argv[3]
+    except IndexError:
+        logger.error("Missing action argument")
+        return 1
+    else:
+        if action not in cli_tm_allowed_actions:
+            logger.error(
+                "Unrecognized action, '{}', allowed actions are: {}",
+                action,
+                cli_tm_allowed_actions,
+            )
+            return 1
+        elif action == "kill":
+            # FIXME: we need to implement the gritty method of killing all the
+            # tool meisters, locally and remotely, and ensuring they are all
+            # properly shut down.
+            return 0
```

Comment:
Although these commands aren't (at least, yet) using `click`, this is a very click-like division, with the very simple CLI wrapper and all logic in a library module.

I like this logic, as the `pbench-tool-meister-start`/etc. modules aren't pytest-able. I suppose that whether we really want/need to test the command line parsing setup is debatable; but we could. In any case, if we do switch these to click, this is how it would look, and that makes some sense to me.

Reply:
We could replace the logger calls with "interface" exceptions which the caller of main can just catch with a `try`/`except` block and do the honors of how to notify the CLI context.

Reply:
For the `.error` and `.exception` CLI log points, that might be a great idea. A lot of the tool meister logging is `.debug`, with some `.info` and `.warning`, none of which are terminal, so an exception model doesn't really work there.

We do need to move the agent away from using the logger for user feedback; though I'd rather not tackle that here.

Reply:
Yes, with click the trivial launch wrapper is generated; but all of the parameter handling is inside the `main` called by that function, just as it is here. While my intent was to move towards a `main` that can just instantiate an object and tickle a method or two, the specific intent here was to minimize the code flow so it can be reviewed as a lift-and-shift. (And that's messy enough to deal with.)